sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/ConfigProperties.java

```diff
@@ -33,7 +44,18 @@ public interface ConfigProperties {
   Boolean getBoolean(String name);
 
   /**
-   * Returns a integer-valued configuration property.
+   * Returns a boolean-valued configuration property.
+   *
+   * @return a Boolean-valued configuration property or {@code defaultValue} if a property with
+   *     {@code name} has not been configured.
+   * @throws ConfigurationException if the property is not a valid string.
+   */
+  default Boolean getBoolean(String name, Boolean defaultValue) {
```

Comment:
```suggestion
  @Nullable
  default Boolean getBoolean(String name, @Nullable Boolean defaultValue) {
```

or 

```suggestion
  default boolean getBoolean(String name, boolean defaultValue) {
```

Reply:
Hmmm....this is super interesting. Do you think we should do that for all the primitive/boxable types?
I worry a little in the inconsistency/mismatch between the primitive one, but I think this is an improvement. I'm gonna go for it for all of them. ❤️